shell/layouts/default.vue

```diff
@@ -194,224 +209,101 @@ export default {
      */
     clusterAndRouteReady() {
-      return this.clusterReady &&
+      return (
+        this.clusterReady &&
         this.clusterId === this.$route?.params?.cluster &&
-        this.currentProduct?.name === getProductFromRoute(this.$route);
-    },
-
-  },
-
-  watch: {
-    counts(a, b) {
-      if ( a !== b ) {
-        this.queueUpdate();
-      }
-    },
-
-    allSchemas(a, b) {
-      if ( a !== b ) {
-        this.queueUpdate();
-      }
-    },
-
-    allNavLinks(a, b) {
-      if ( a !== b ) {
-        this.queueUpdate();
-      }
+        this.currentProduct?.name === getProductFromRoute(this.$route)
+      );
     },
 
-    favoriteTypes(a, b) {
-      if ( !isEqual(a, b) ) {
-        this.queueUpdate();
-      }
-    },
-
-    locale(a, b) {
-      if ( !isEqual(a, b) ) {
-        this.getGroups();
-      }
-    },
-
-    productId(a, b) {
-      if ( !isEqual(a, b) ) {
-        // Immediately update because you'll see it come in later
-        this.getGroups();
-      }
-    },
-
-    clusterId(a, b) {
-      if ( !isEqual(a, b) ) {
-        // Store the last visited route when the cluster changes
-        this.setClusterAsLastRoute();
-      }
-    },
-
-    namespaceMode(a, b) {
-      if ( !isEqual(a, b) ) {
-        // Immediately update because you'll see it come in later
-        this.getGroups();
-      }
-    },
-
-    namespaces(a, b) {
-      if ( !isEqual(a, b) ) {
-        // Immediately update because you'll see it come in later
-        this.getGroups();
-      }
-    },
-
-    clusterReady(a, b) {
-      if ( !isEqual(a, b) ) {
-        // Immediately update because you'll see it come in later
-        this.getGroups();
-      }
-    },
-
-    product(a, b) {
-      if ( !isEqual(a, b) ) {
-        // Immediately update because you'll see it come in later
-        this.getGroups();
-      }
-    },
-
-    async currentProduct(a, b) {
-      if ( !isEqual(a, b) ) {
-        if (a.inStore !== b.inStore || a.inStore !== 'cluster' ) {
-          const route = {
-            name:   'c-cluster-product',
-            params: {
-              cluster: this.clusterId,
-              product: a.name,
-            }
-          };
+    loadedProducts() {
+      const loadProducts = this.isExplorer ? [EXPLORER] : [];
+      const currentProduct = this.productId; // Use the productId computed property as a dependency to trigger updates when the product changes.
 
-          await this.$store.dispatch('prefs/setLastVisited', route);
+      if (this.isExplorer) {
+        for (const product of this.activeProducts) {
+          if (product.inStore === 'cluster') {
+            addObject(loadProducts, product.name);
+          }
         }
       }
-    },
-
-    $route(a, b) {
-      this.$nextTick(() => this.syncNav());
-    },
-
-  },
-
-  async created() {
-    this.queueUpdate = debounce(this.getGroups, 500);
```

Comment:
I removed the debounce because I think it's no longer necessary because we now limit how often are schemas are updated, so that should have a similar effect as the debounce, unless I'm mistaken.

Reply:
I think the blank home page was a separate bug because after I rebased on master, the Home page worked again.